services/elasticache/apis/v1alpha1/snapshot.go

```diff
@@ -0,0 +1,87 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+// Code generated by ack-generate. DO NOT EDIT.
+
+package v1alpha1
+
+import (
+	ackv1alpha1 "github.com/aws/aws-controllers-k8s/apis/core/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// SnapshotSpec defines the desired state of Snapshot
+type SnapshotSpec struct {
+	CacheClusterID     *string `json:"cacheClusterID,omitempty"`
+	KMSKeyID           *string `json:"kmsKeyID,omitempty"`
+	ReplicationGroupID *string `json:"replicationGroupID,omitempty"`
+	// +kubebuilder:validation:Required
+	SnapshotName       *string `json:"snapshotName"`
+	SourceSnapshotName *string `json:"sourceSnapshotName,omitempty"`
```

Comment:
:+1: 

I think eventually I'd like to inject some code comments here for "additional spec fields" to indicate that these are fields that come from non-standard-create operations. I'd also like the ability to inject some docstring here to help inform the Kubernetes user of these CRDs how to use these fields.

Reply:
Agreed on this I will add this as part of https://github.com/aws/aws-controllers-k8s/issues/447